docs/adr/adr-0018-how-to-give-feedback-after-a-user-action.md

```diff
@@ -0,0 +1,414 @@
+const dismissToast = document.querySelector('cc-toaster').show({
+  message: 'Something went wrong while perfoming action',
+  intent: 'danger'
+});
+
+// Dismissing a toast is done by using the function returned by the `show()` method
+dismissToast();
+```
+
+This final implementation takes the best from the previous attempts.
+
+We removed the magic happening when the `<cc-toaster>` element was added to the DOM.
+As a counterpart, developers need to code the glue between their application and the toaster.
+
+We kept the ability to programmatically dismiss a toast, but without the need to maintain a unique identifier for the toast.
+
+### Wrapping up
+
+First, we wanted the integration of the toaster to be done with the less code as possible.
+We figured out that it introduced a bit too much of magic.
+So we came back to something that needs a bit of glue, and in fact, it provides greater flexibility for developers.
+
+We believe that the integration should be done with decoupling in mind and a nice way to do that is to use DOM event.
+The code bellow shows how we integrate the `cc-toaster` in the Clever Console:
+
+```javascript
+const toaster = document.querySelector('cc-toaster');
+window.addEventListener('cc:notify', ({ detail: notification }) => {
+  toaster.show(notification);
+});
+
+export function notify(node, message, intent) {
+  const toast = { message, intent };
+  node.dispatchEvent(new CustomEvent('cc:notify', { detail: toast, bubbles: true, composed: true }));
+}
+```
+
+## Accessibility
+
+When we design a new component, we always try to be as inclusive as possible.
+Meaning that the sooner we think about accessibility, the better.
+
+Some great articles (listed in the resources section) helped us understand all the issues related to the accessibility of the `toast` pattern.
+
+We understood that the toast pattern comes, by design, with a lot of accessibility issues:
+
+* toasts are displayed a short amount of time and some user may miss the notification
+* they are placed in the corner of the screen and therefore, people using a screen magnifier may totally miss them.
+* they reveal their intention by using colors (green for success, red for error) which causes issues for colorblindness people.
+* they may contain buttons which should be focusable with keyboard, but the focus may be lost when the toast disappears.
+  And if these actions are critical, user may miss important actions.
+* screen readers expect dynamic content to be inserted inside a _live region element_ which has sporadic support.
+
+Some issues could be addressed, but we quickly understood it will come with some compromises.
+Here is what we implemented.
+
+### Toast color
+
+Color is not enough to reveal the notification intention, so we added an icon to help colorblind users to understand it.
+We also added an `alt` text on that icon so that people using screen reader are aware of the notification intention.
```

Comment:
I would even add that the alt text is translated in english and french depending on the selected language. I propose this because most of the off the shelf solution don't do this tiny detail ;-)